test/e2e/upgrade_test.go

```diff
@@ -176,11 +173,10 @@ func (tc *testContext) scaleWMCODeployment(desiredReplicas int32) error {
 // deployWindowsWorkloadTester tests if the Windows Webserver deployment is available.
 // This is achieved by creating a Job object that continuously curls the webserver every 5 seconds.
 func (tc *testContext) deployWindowsWorkloadTester() error {
-	// Get Windows Webserver deployment
-	deployment, err := tc.kubeclient.AppsV1().Deployments(tc.workloadNamespace).Get(context.TODO(), "win-webserver-deployment",
-		metav1.GetOptions{})
+	// Create a Windows Webserver deployment
+	deployment, err := tc.deployWindowsWebServer("win-webserver", nil)
```

Comment:
From my reading this breaks the upgrade test. I don't think we are actually testing the upgrade scenario if we create the deployment here. How are we ensuring the workload is functioning across a WMCO "upgrade".

Reply:
@VaishnaviHire please confirm.

Reply:
[We deploy the workload before scaling down the operator and configuring it for upgrade](https://github.com/openshift/windows-machine-config-operator/blob/602f4f140a57582eb0fc5b08c29795fc6e922cbe/test/e2e/upgrade_test.go#L38-L44)

Reply:
Ah I see. Sorry I missed that. What you then need to do is not create the workload in `deployWindowsWorkloadTester()`. Or change the name of the function to something else, maybe `deployWindowsWorkloadAndTester()`. 

I am also not a huge fan of not cleaning up resources given your `Everything should be destroyed once we tear down the cluster.`. While you are correct it is not a good pattern to follow. There could be situations where a dangling resource could prevent a cluster from being torn down gracefully.